myokit/formats/sbml/_importer.py

```diff
@@ -35,334 +36,1065 @@ def __init__(self):
-                'Converting name <' + org_name + '> to <' + name + '>.')
-        return name
+        return True
 
-    def _convert_unit(self, unit):
+    def supports_component(self):
         """
-        Converts an SBML unit to a myokit one using the lookup tables generated
-        when parsing the XML file.
+        Returns a bool indicating if component import is supported.
         """
-        if unit in self.units:
-            return self.units[unit]
-        elif unit in unit_map:
-            return unit_map[unit]
-        else:      # pragma: no cover
-            raise SBMLError('Unit not recognized: ' + str(unit))
+        return True
 
-    def info(self):
-        return info
+    def parse_file(self, path):
+        """
+        Parses the SBML file at ``path`` and returns a myokit
+        model.
+        """
+        # Read file
+        try:
+            tree = ET.parse(path)
+        except Exception as e:
+            raise SBMLError('Unable to parse XML: ' + str(e))
 
-    def model(self, path, bind_time=True):
+        # Parse content
+        return self.parse(tree.getroot())
+
+    def parse_string(self, text):
         """
-        Returns a :class:myokit.Model based on the SBML file provided.
+        Parses the SBML XML in the string ``text`` and returns
+        a myokit model.
+        """
+        # Read string
+        try:
+            root = ET.fromstring(text)
+        except Exception as e:
+            raise SBMLError('Unable to parse XML: ' + str(e))
 
-        Arguments:
+        # Parse content
+        return self.parse(root)
 
-        ``path``
-            The path to the SBML file.
-        `` bind_time``
-            If set to ``True`` (default), a variable called "time" will be
-            created and bound to `time`.
+    def parse(self, root):
+        """
+        Parses and a SBML document rooted in the given elementtree
+        element.
+        """
+        try:
+            return self._parse_model(root)
+        except myokit.formats.sbml._importer.SBMLError as e:
+            raise SBMLError(str(e))
+        except myokit.formats.mathml.MathMLError as e:
+            raise SBMLError(str(e))
 
+    def _parse_model(self, root):
+        """
+        Returns a :class:myokit.Model based on the SBML file provided.
         """
         # Get logger
         log = self.logger()
 
-        # Parse xml file
-        path = os.path.abspath(os.path.expanduser(path))
-        tree = ET.parse(path)
-        root = tree.getroot()
-
-        # get SBML namespace
-        sbml_version = self._get_sbml_version(root)
-        self.ns = self._get_namespaces(sbml_version, log)
-
-        # Get model node
-        xmodel = root[0]
-        if xmodel.get('name'):
-            name = str(xmodel.get('name'))
-        elif xmodel.get('id'):
-            name = str(xmodel.get('id'))
-        else:
+        # Check whether file has SBML 3.2 namespace
+        ns = self._getNamespace(root)
```

Comment:
Myokit uses `CamelCase` for classes, and `whats_apparently_called_snake_case` for module, method and variable names.

(With the exception being publically visible module-level constants, which are `WRITTEN_LIKE_THIS_FOR_EXTRA_UGLINESS`, but I don't think you have any of those)

Reply:
Oki doki. I played around with libsmbl too much so now my naming is all over the place. :D